ion/type_test.go

```diff
@@ -19,3 +19,93 @@ func TestIntSizeToString(t *testing.T) {
 		}
 	}
 }
+
+func TestIsScalar(t *testing.T) {
+	if IsScalar(NoType) {
```

Comment:
To eliminate some of the copy/pasting, I suggest making an array of the scalar types and looping over them. The same could be done for `TestIsContainer`.

I'm new to Go, so if there are reasons that's ill advised, please let me know.

Reply:
Done.